### PR TITLE
setup coco attestation on migration

### PR DESCRIPTION
**Pull request by lucidd**
## What does this PR change?

This makes sure coco attestation is usable right after migrating from an old installation.

## Test coverage
- No tests

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24323

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by mbussolotto — APPROVED** [review]

File: mgradm/shared/podman/podman.go [hunk 1]
```diff
@@ -342,6 +342,30 @@ func RunPostUpgradeScript(serverImage string) error {
...
+		return utils.Errorf(err, L("failed to create temporary directory"))
+	}
+	containerName := "uyuni-coco-dbconfig"
+	extraArgs := []string{
+		"-v", scriptDir + ":/var/lib/uyuni-tools/",
+		"--security-opt", "label=disable",
+	}
+	scriptName, err := adm_utils.GenerateCocoDBConfigScript(scriptDir, user, dbname, ip)
+	if err != nil {
+		return utils.Errorf(err, L("cannot generate coco db config script"))
```
Comment (cbosdo) [inline mgradm/shared/podman/podman.go:359]:
```suggestion
		return utils.Errorf(err, L("cannot generate confidential container database configuration script"))
```

File: mgradm/shared/templates/pgsqlVersionUpgradeScriptTemplate.go [hunk 2]
```diff
@@ -46,6 +46,9 @@ su -s /bin/bash - postgres -c "initdb -D /var/lib/pgsql/data --locale=$POSTGRES_
 echo "Successfully initialized new postgresql $NEW_VERSION database."
 su -s /bin/bash - postgres -c "pg_upgrade --old-bindir=/usr/lib/postgresql$OLD_VERSION/bin --new-bindir=/usr/lib/postgresql$NEW_VERSION/bin --old-datadir=/var/lib/pgsql/data-pg$OLD_VERSION --new-datadir=/var/lib/pgsql/data $FAST_UPGRADE"
 
+cp /var/lib/pgsql/data-pg$OLD_VERSION/pg_hba.conf /var/lib/pgsql/data
+cp /var/lib/pgsql/data-pg${OLD_VERSION}/postgresql.conf /var/lib/pgsql/data/
```
Comment (cbosdo) [inline mgradm/shared/templates/pgsqlVersionUpgradeScriptTemplate.go:50]:
```suggestion
cp /var/lib/pgsql/data-pg$OLD_VERSION/pg_hba.conf /var/lib/pgsql/data
cp /var/lib/pgsql/data-pg$OLD_VERSION/postgresql.conf /var/lib/pgsql/data/
```

We don't use curly braces around variables in that script, better use a similar style everywhere.

**Review by cbosdo — COMMENTED** [review]
Cosmetic suggestions, but the rest LGTM

**Review by cbosdo — APPROVED** [review]

File: mgradm/cmd/migrate/shared/flags.go [hunk 3]
```diff
@@ -15,6 +16,7 @@ import (
 type MigrateFlags struct {
 	Image          types.ImageFlags `mapstructure:",squash"`
 	DbUpgradeImage types.ImageFlags `mapstructure:"dbupgrade"`
+	Coco           shared.CocoFlags
```
Comment (cbosdo) [inline mgradm/cmd/migrate/shared/flags.go:19]: You don't add the parameters to set those flags to the migrate command

Comment (cbosdo) [reply mgradm/cmd/migrate/shared/flags.go:19]: Added

File: mgradm/cmd/migrate/podman/utils.go [hunk 4]
```diff
@@ -93,5 +95,25 @@ func migrateToPodman(globalFlags *types.GlobalFlags, flags *podmanMigrateFlags,
...
+
+	dbPort, err := strconv.Atoi(inspectedValues["db_port"])
+	if err != nil {
+		return utils.Errorf(err, L("error %s is not a valid port number."), inspectedValues["db_port"])
+	}
+
+	err = coco.Upgrade(flags.Coco.Image, flags.Image,
+		dbPort, inspectedValues["db_name"], inspectedValues["db_user"], inspectedValues["db_password"])
+	if err != nil {
+		return utils.Errorf(err, L("error upgrading confidential computing service."))
```
Comment (cbosdo) [inline mgradm/cmd/migrate/podman/utils.go:115]: Upgrade is not enough: we need to scale after

**Pull request closed without merging**